keel-slack/src/main/kotlin/com/netflix/spinnaker/keel/slack/NotificationEventListener.kt

```diff
@@ -306,10 +370,25 @@ class NotificationEventListener(
 
     environments.flatMap { it.notifications }
       .filter { it.type == NotificationType.slack }
+      .filter { translateFrequencyToEvents(it.frequency).contains(type) }
       .groupBy { it.address }
       .forEach { (channel, _) ->
         handler.sendMessage(message, channel)
       }
   }
+
+
```

Comment:
The way you've structured this function (and used it above) is kind of hard to read. 

I think what you're saying is "filter the list of notifications to only the ones I should send based on the frequency" - but even typing this I'm not sure. I'd recommend restructuring this so it reads easier when you're using it. Maybe that means having a "shouldSend" function, or something that reads like `.filter { notification -> notification.shouldSend }`

Reply:
I plan to refactor this class. Do you feel it's blocking from merging it? The reason I'm asking is I want to release this feature soon and I'd like to release it with the freq options available.